cmake/DD4hepConfig.cmake.in

```diff
@@ -21,12 +21,24 @@ if (_lastdir STREQUAL "cmake")
   get_filename_component(_thisdir "${_thisdir}" PATH)
 endif()
 
-set ( DD4hep_DIR         "${_thisdir}" )
-set ( DD4hep_ROOT        "${_thisdir}" )
-set ( DD4hep_VERSION     "@DD4hep_VERSION@" )
-set ( DD4hep_SOVERSION   "@DD4hep_SOVERSION@" )
-set ( DD4HEP_USE_GEANT4  "@DD4HEP_USE_GEANT4@" )
-set ( DD4HEP_USE_XERCESC "@DD4HEP_USE_XERCESC@" )
+set ( DD4hep_DIR                "${_thisdir}" )
+set ( DD4hep_ROOT               "${_thisdir}" )
+set ( DD4hep_VERSION            "@DD4hep_VERSION@" )
+set ( DD4hep_SOVERSION          "@DD4hep_SOVERSION@" )
+set ( DD4HEP_USE_GEANT4         "@DD4HEP_USE_GEANT4@" )
+set ( DD4HEP_USE_XERCESC        "@DD4HEP_USE_XERCESC@" )
+set ( DD4hep_BUILD_CXX_STANDARD @CMAKE_CXX_STANDARD@ )
+
+# figure out if including build has compatible standard set
+if("${CMAKE_CXX_STANDARD}" STREQUAL "")
+  set ( CMAKE_CXX_STANDARD  "${DD4hep_BUILD_CXX_STANDARD}" )
+else()
+  if (NOT "${CMAKE_CXX_STANDARD}" EQUAL "${DD4hep_BUILD_CXX_STANDARD}")
+    message( SEND_ERROR "CMAKE_CXX_STANDARD is set to C++${CMAKE_CXX_STANDARD}\
+             but C++${DD4hep_BUILD_CXX_STANDARD} is required.")
+  endif()
+endif()
+
 set ( CMAKE_CXX_STANDARD  @CMAKE_CXX_STANDARD@ )
```

Comment:
You would want to remove this line as well, right?

Reply:
Absolutely, I guess I missed that.